WPFHospitalEditor/EquipmentRelocation.xaml.cs

```diff
@@ -0,0 +1,167 @@
+
+            if (AmountIsValid() && timeInterval.IsValid())
+            {
+                EquipmentRelocationDto eqRelDto = CreateEquipmentRelocationDto(timeInterval);
+                List<int> unavailableRooms = roomServerController.GetUnavailableRoomsIdsInTimeInterval(eqRelDto).ToList();
+                if (unavailableRooms.Count > 0)
+                {
+                    ShowAlternativeRelocationAppointments(unavailableRooms);
+                }
+                else
+                {
+                    ScheduleRelocation(eqRelDto);
+                }
+            }
+
+        }
+
+        private void ShowAlternativeRelocationAppointments(List<int> unavailableRooms)
+        {
+            AlternativeRelocationAppointments newWindow =
+            new AlternativeRelocationAppointments(unavailableRooms[0], this);
+            newWindow.Show();
+        }
+
+        private void ScheduleRelocation(EquipmentRelocationDto eqRelDto)
+        {
+
+            List<int> doctors = doctorServerController.GetDoctorsByRoomsAndShifts(eqRelDto).ToList();
+            foreach (int doctorId in doctors)
+            {
+                startDate = eqRelDto.TimeInterval.Start;
+                while (startDate < endDate)
+                {
+                    examinationServerController.ScheduleExamination(startDate, doctorId, 2);
+                    startDate = startDate.AddMinutes(30);
+                }
+            }
+            MessageBox.Show("Relocation is successfully scheduled!", "");
+            this.Close();
+        }
+
+        private EquipmentRelocationDto CreateEquipmentRelocationDto(TimeInterval timeInterval)
+        {
+            EquipmentRelocationDto equipmentRelocationDto = new EquipmentRelocationDto()
+            {
+                SourceRoomId = (int)roomSearchComboBox.SelectedItem,
+                DestinationRoomId = destinationRoomId,
+                Amount = int.Parse(equipmentAmount.Text),
+                EquipmentType = relocationEquipmentName,
+                TimeInterval = timeInterval
+
+            };
+            return equipmentRelocationDto;
+        }
+
+        private void FillObservableCollection()
+        {
+            roomsWithEquipmentAmount = new ObservableCollection<RoomWithChosenEquipmentAmount>();
+            List<Room> rooms = roomServerController.getRoomsByEquipmentType(relocationEquipmentName).ToList();
+            List<EquipmentDto> equipments = equipmentServerController.GetEquipmentByType(relocationEquipmentName).ToList();
+            foreach (Room room in rooms)
+            {
+                foreach (EquipmentDto eqdto in equipments)
+                {
+                    if (eqdto.RoomId == room.Id && room.Id != destinationRoomId)
+                    {
+                        roomsWithEquipmentAmount.Add(new RoomWithChosenEquipmentAmount(room.Id, room.Name, eqdto.Quantity));
+                    }
+                }
```

Comment:
Looks like you can use only for with equipmentDTOs because there is information about roomId.